micrometer-core/src/main/java/io/micrometer/core/instrument/binder/kafka/KafkaMetrics.java

```diff
@@ -132,6 +139,17 @@ else if (tags.size() == meterTags.size())
         }
     }
 
+    private boolean differentClient(List<Tag> tags) {
+        boolean anotherClient = false, anotherVersion = false;
+        for (Tag tag : tags) {
+            if (tag.getKey().equals(CLIENT_ID_TAG_NAME))
+                if (!clientId.equals(tag.getValue())) anotherClient = true;
+            if (tag.getKey().equals(KAFKA_VERSION_TAG_NAME))
+                if (!kafkaVersion.equals(tag.getValue())) anotherVersion = true;
```

Comment:
I'm wondering how there would be multiple versions of the kafka client at the same time without some classloader trickery. That said, we can still guard against this - just wondering if it is a case you are expecting.

Reply:
True, I think I went to far with this hehe. I was considering two apps same client id with diff version but that would happen on different jvms. I will keep it simple without version and add an IT.